python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -74,6 +73,7 @@ class Node(object):
     thread-safe usage.
     """
     __metaclass__ = abc.ABCMeta
+    __slots__ = ('_index_cache_dict', '_parent', '_read_lock', '_write_lock', '_dirty', '_registry', '_data_dict')
```

Comment:
I agree in general that for our code, preventing dynamic attributes is a good idea. However in practice many `GangaObject` subclasses create `self.temp_foo` variables outside of the schema for tracking internal state. Also, further down, `__dict__` is added to `__slots__` anyway which re-allows dynamic attributes to be added. I'd be surprised if this gave more than a few percent improvement, especially given that most of the data in a `GangaObject` is in `_data` anyway and that gets quite large.